app/src/main/java/org/wikipedia/language/LangLinksActivity.kt

```diff
@@ -210,7 +210,7 @@ class LangLinksActivity : BaseActivity() {
             languageEntries.clear()
             for (entry in originalLanguageEntries) {
                 val languageCode = entry.wikiSite.languageCode
-                val canonicalName = app.languageState.getAppLanguageCanonicalName(languageCode).orEmpty()
+                val canonicalName = viewModel.getCanonicalName(languageCode) ?: app.languageState.getAppLanguageCanonicalName(languageCode).orEmpty()
```

Comment:
Here this also fixes an issue that existed silently: the language names were not searchable properly -- they were only searchable by the English name, instead of by the name given by the `siteinfo` call.